src/OddJobs/Job.hs

```diff
@@ -261,7 +261,7 @@ withDbConnection :: (HasJobRunner m)
                  -> m a
 withDbConnection action = do
   pool <- getDbPool
-  withResource pool action
+  withRunInIO $ \runInIO -> withResource pool $ runInIO . action
```

Comment:
why was `MonadBaseControl` removed from `Data.Pool`. It was pretty nifty. Any way to write a wrapper to remove this `runInIO` noise all over the place?

Reply:
> why was `MonadBaseControl` removed from `Data.Pool`. It was pretty nifty. Any way to write a wrapper to remove this `runInIO` noise all over the place?

You may want to ask @arybczak directly

Reply:
It was removed because some people want `MonadBaseControl` and some want `MonadUnliftIO`, can't please everyone.

You can always define a local wrapper for `withResource` that uses `MonadUnliftIO` :slightly_smiling_face:

Reply:
Anyone interested in modifying this PR to create a user a wrapped on `withResource` with the following signature:

```haskell
withResource :: MonadUnliftIO m => Pool a -> (a -> m b) -> m b
```

Also, should that be `MonadIO` or `MonadUnliftIO`?

Reply:
no worries saura I'll do this :+1: